### PR TITLE
collection: create public and professional views 

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Adds public brief view.
* Adds public routing factory and route definition.
* Adds professional brief and detailed views.
* Adds the application initialisation system.
* Adds some license on files.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):
* https://github.com/rero/rero-ils-ui/pull/1225

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/app.module.ts [hunk 1]
```diff
@@ -106,21 +118,17 @@ import { PersonDetailViewComponent } from './record/detail-view/person-detail-vi
...
 import { AppInitService } from './service/app-init.service';
 
 /** Init application factory */
 export function appInitFactory(appInitService: AppInitService) {
   return () => appInitService.load();
 }
+import { SharedPipesModule } from './shared/shared-pipes.module';
+import {
+  HoldingItemInCollectionComponent
+} from './record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component';
```
Comment (zannkukai) [inline projects/admin/src/app/app.module.ts:131]: Why these imports are not placed with others ?

File: projects/admin/src/app/class/items-typeahead.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    const pid = url.pop();
+
+    return this._recordService.getRecord(options.type, pid, 1)
+    .pipe(
+      switchMap((record: any) => {
+        return this._recordService.getRecord('documents', record.metadata.document. pid)
+        .pipe(
+          map((document: any | Error) =>
+            `<span class="pt-0">
+              <h6 class="my-0">${this._mainTitlePipe.transform(document.metadata.title)}</h6>
```
Comment (zannkukai) [inline projects/admin/src/app/class/items-typeahead.ts:64]: `h6` isn't a little bit overkill ?

File: projects/admin/src/app/record/brief-view/collection-brief-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  <div class="card-text">
+    <div id="collection-teacher" *ngIf="record.metadata.teachers">
+      <ng-container *ngFor="let teacher of record.metadata.teachers; let last = last">
+      {{ teacher.name }}
+      {{ last ? '' : ', ' }}
+      </ng-container>
+    </div>
+    <div
+      id="collection-start-end-date"
+      *ngIf="record.metadata.description"
```
Comment (zannkukai) [inline projects/admin/src/app/record/brief-view/collection-brief-view.component.ts:37]: maybe move `*ngIf` statement as top attribute ?

File: projects/public-search/src/app/collection-brief/collection-brief.component.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  <div class="card-text">
+    <div id="collection-teacher" *ngIf="record.metadata.teachers">
+      <ng-container *ngFor="let teacher of record.metadata.teachers; let last = last">
+      {{ teacher.name }}
+      {{ last ? '' : ', ' }}
+      </ng-container>
+    </div>
+    <div
+      id="collection-start-end-date"
+      *ngIf="record.metadata.description"
```
Comment (zannkukai) [inline projects/public-search/src/app/collection-brief/collection-brief.component.html:33]: `*ngIf` as first attribute

File: projects/public-search/src/app/routes/base-route.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+export class BaseRoute {
+
+  private _translateService: TranslateService;
+
+  /**
+   * Constructor
+   * @param _translateService - TranslateService
+   */
+  protected constructor(
+    translateService: TranslateService
```
Comment (zannkukai) [inline projects/public-search/src/app/routes/base-route.ts:31]: check doc string param: either remove the '_' either add it as constructor parameter

File: projects/public-search/src/app/routes/base-route.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  ) {
+    this._translateService = translateService;
+  }
+
+  /**
+   * Route matcher
+   * @param url - any
+   * @param type - string
```
Comment (zannkukai) [inline projects/public-search/src/app/routes/base-route.ts:39]: move `type` to `types` and change doc string

File: projects/public-search/src/app/routes/base-route.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    const urlType = url.slice(-1).pop();
+    if (types.some(x => x === urlType.path)) {
+      return this.matchedUrl(url);
+    }
+    return null;
+  }
+
+  /**
+   * Matched url
+   * @param url - UrlSegment
```
Comment (zannkukai) [inline projects/public-search/src/app/routes/base-route.ts:51]: Array<UrlSegment>

File: projects/public-search/src/app/routes/route-factory.service.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+  /**
+   * Constructor
+   * @param _injector - Injector
+   * @param _router - Router
+   */
+  constructor(
+    private _router: Router,
+    private _routeCollectionService: RouteCollectionService
+  ) {}
```
Comment (zannkukai) [inline projects/public-search/src/app/routes/route-factory.service.ts:35]: Check doc string

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/app.module.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/admin/src/app/app.module.ts:131]: done

File: projects/admin/src/app/class/items-typeahead.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/admin/src/app/class/items-typeahead.ts:64]: done

File: projects/admin/src/app/record/brief-view/collection-brief-view.component.ts (hunk 3 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/brief-view/collection-brief-view.component.ts:37]: done

File: projects/admin/src/app/record/detail-view/collection-detail-view/collection-items/collection-items.component.html [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  <ul class="list-unstyled">
+    <ng-container *ngFor="let item of items; let isOdd=odd">
+      <li class="py-1" [ngClass]="{ 'bg-light': isOdd }" *ngIf="item.pid | getRecord: 'items' | async as itemData">
+        <ng-container
+          *ngIf="itemData.metadata.document.pid | getRecord:'documents':'object':null:{Accept: 'application/rero+json'} | async as document"
+        >
+          <h5>
+            <a
+              [routerLink]="['/records','documents', 'detail', itemData.metadata.document.pid]"
+            >
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/collection-detail-view/collection-items/collection-items.component.html:28]: done

File: projects/admin/src/app/record/detail-view/collection-detail-view/collection-items/collection-items.component.html [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+              <span class="font-weight-bold" translate>Contribution</span>:
+              <ng-container
+                class="list-inline-item"
+                *ngFor="let contribution of document.metadata.contribution; let last = last; let i = index"
+              >
+                <span *ngIf="!contribution.agent.pid" [innerHTML]="contribution | contributionFormat: true"></span>
+                <a
+                *ngIf="contribution.agent.pid"
+                [innerHTML]="contribution | contributionFormat: true"
+                [routerLink]="['/records', 'persons', 'detail', contribution.agent.pid]">
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/collection-detail-view/collection-items/collection-items.component.html:43]: done

File: projects/admin/src/app/record/detail-view/collection-detail-view/collection-items/collection-items.component.html [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+            </div>
+            <div *ngIf="itemData.metadata.call_number">
+              <span class="font-weight-bold" translate>Call number</span>:
+              {{ itemData.metadata.call_number }}
+              <span *ngIf="itemData.metadata.second_call_number">
+              | {{ itemData.metadata.second_call_number }}
+              </span>
+            </div>
+            <ng-container *ngIf="itemData.metadata.location.pid | getRecord:'locations' | async as location">
+              <div>
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/collection-detail-view/collection-items/collection-items.component.html:64]: done

File: projects/public-search/src/app/collection-brief/collection-brief.component.html (hunk 4 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/collection-brief/collection-brief.component.html:33]: done

File: projects/public-search/src/app/routes/base-route.ts (hunk 6 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/routes/base-route.ts:39]: done

File: projects/public-search/src/app/routes/base-route.ts (hunk 5 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/routes/base-route.ts:31]: done

File: projects/public-search/src/app/routes/base-route.ts (hunk 7 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/routes/base-route.ts:51]: done

File: projects/public-search/src/app/routes/route-factory.service.ts (hunk 8 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/routes/route-factory.service.ts:35]: done

**Review by benerken — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts [hunk 12]
```diff
@@ -14,17 +14,17 @@
...
-import { BsModalService } from 'ngx-bootstrap/modal';
 import { RecordUiService } from '@rero/ng-core';
-import { forkJoin } from 'rxjs';
-import { first } from 'rxjs/operators';
+import { BsModalService } from 'ngx-bootstrap/modal';
 import { ItemsService } from 'projects/admin/src/app/service/items.service';
 import { RecordPermissionService } from 'projects/admin/src/app/service/record-permission.service';
 import { UserService } from 'projects/admin/src/app/service/user.service';
+import { forkJoin } from 'rxjs';
+import { first } from 'rxjs/operators';
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:25]: These imports are not used next in the file.

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+@Component({
+  selector: 'admin-holding-item-in-collection',
+  templateUrl: './holding-item-in-collection.component.html'
+})
+export class HoldingItemInCollectionComponent implements OnInit {
+
+  /** Item pid */
+  @Input() itemPid: string;
+
+  @Input() class: string;
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts:32]: Doc block is missing.

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+  /** Collections */
+  collections: [];
+
+  constructor(private _recordService: RecordService) { }
+
+  ngOnInit() {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts:39]: Same here

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+    this.isItemInCollection(this.itemPid);
+  }
+
+  private isItemInCollection(itemPid: string) {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts:43]: Doc block + return type

File: projects/public-search/src/app/error/error-page.component.ts [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { marker } from '@biesbjerg/ngx-translate-extract-marker';
+import { map } from 'rxjs/operators';
+
+export function _(str: string) {
+  return marker(str);
+}
```
Comment (sebastiendeleze) [inline projects/public-search/src/app/error/error-page.component.ts:25]: I think you can use this import directly: 
```
import { marker as _ } from '@biesbjerg/ngx-translate-extract-marker';
```

File: projects/public-search/src/app/guard/collection-access.guard.ts [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+  providedIn: 'root'
+})
+export class CollectionAccessGuard implements CanActivate {
+
+  constructor(
+    private _appConfigService: AppConfigService,
+    private _router: Router
+  ) {}
+
+  canActivate(
```
Comment (sebastiendeleze) [inline projects/public-search/src/app/guard/collection-access.guard.ts:33]: Doc blocks are missing + return type.

**Review by sebastiendeleze — CHANGES_REQUESTED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts (hunk 12 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:25]: This is used on line 99

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts (hunk 13 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts:32]: done

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts (hunk 15 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts:43]: done

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts (hunk 14 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/document-detail-view/holding/holding-item-in-collection/holding-item-in-collection.component.ts:39]: done

File: projects/public-search/src/app/error/error-page.component.ts (hunk 16 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/error/error-page.component.ts:25]: done

File: projects/public-search/src/app/guard/collection-access.guard.ts (hunk 17 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/guard/collection-access.guard.ts:33]: done

**Review by sebastiendeleze — APPROVED** [review]

**Pull request merged**